apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -18,26 +20,61 @@ export class GetActiveIntegrations {
     private integrationRepository: IntegrationRepository,
     private selectIntegration: SelectIntegration,
     private environmentRepository: EnvironmentRepository,
-    private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations
+    private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations,
+    private getFeatureFlag: GetFeatureFlag
   ) {}
 
   async execute(command: GetActiveIntegrationsCommand): Promise<GetActiveIntegrationResponseDto[]> {
-    const activeIntegration = await this.getDecryptedIntegrationsUsecase.execute(
-      GetDecryptedIntegrationsCommand.create({
-        organizationId: command.organizationId,
-        userId: command.userId,
-        active: true,
-      })
-    );
+    const environments = await this.environmentRepository.findOrganizationEnvironments(command.organizationId);
 
-    if (!activeIntegration.length) {
+    const activeIntegrations = await this.getActiveIntegrations(command, environments);
+
+    if (!activeIntegrations.length) {
       return [];
     }
 
-    const activeIntegrationChannelTypes = this.getDistinctChannelTypes(activeIntegration);
-    const selectedIntegrations = await this.getSelectedIntegrations(command, activeIntegrationChannelTypes);
+    const activeIntegrationChannelTypes = this.getDistinctChannelTypes(activeIntegrations);
+    const selectedIntegrations = await this.getSelectedIntegrations(
+      command,
+      activeIntegrationChannelTypes,
+      environments
+    );
+
+    return this.mapBySelectedIntegration(activeIntegrations, selectedIntegrations);
+  }
 
-    return this.mapBySelectedIntegration(activeIntegration, selectedIntegrations);
+  private async getActiveIntegrations(command: GetActiveIntegrationsCommand, environments: EnvironmentEntity[]) {
+    const isMultiProviderConfigurationEnabled = await this.getFeatureFlag.isMultiProviderConfigurationEnabled(
+      FeatureFlagCommand.create({
+        userId: command.userId,
+        organizationId: command.organizationId,
+        environmentId: command.environmentId,
+      })
+    );
+
+    const activeIntegrationPromises = isMultiProviderConfigurationEnabled
+      ? [
+          await this.getDecryptedIntegrationsUsecase.execute(
+            GetDecryptedIntegrationsCommand.create({
+              organizationId: command.organizationId,
+              environmentId: command.environmentId,
+              userId: command.userId,
+              active: true,
+            })
+          ),
+        ]
+      : environments.map(async (environment) => {
+          return this.getDecryptedIntegrationsUsecase.execute(
+            GetDecryptedIntegrationsCommand.create({
+              environmentId: environment._id,
+              organizationId: command.organizationId,
+              userId: command.userId,
+              active: true,
+            })
+          );
+        });
+
+    return (await Promise.all(activeIntegrationPromises)).flat();
```

Comment:
Why don't we for any case we bring all the integrations by organizationId and for `isMultiProviderConfigurationEnabled=false` we filter the integrations by environmentId?
If worried about decryption cost there is always the chance of take advantage of the caching.

Reply:
What logic we would need to add to `getDecryptedIntegrations`?

Reply:
I hope I am not missing something but let's break it to steps:

> @djabarovgeorge Why don't we for any case we bring all the integrations by organizationId 

This one is manageable when isMultiProviderConfigurationEnabled=true.

> and for `isMultiProviderConfigurationEnabled=false` we filter the integrations by environmentId? 

But this one will not work because if we pass envorinmentId and isMultiProviderConfigurationEnabled=false then the use-case will fetch by the environment. (this is why I iterate through the environment ids in my logic)

The other way around will be adding changes to getDecryptedIntegrations use case,
In order to get all by Organization we will need to not pass environment id, and if we won't pass environment id we would need to add a logit that will avoid extracting the feature flag (because environment will be missing.), maybe add other logic as well, but this one feels even hackier.

Reply:
Not sure if I am missing anything but considering that to the endpoint we always pass `organizationId` and `environmentId` in `GetActiveIntegrationsCommand` as they are injected by JWT in the endpoint, if in the use case we can do:
```
let organizationActiveIntegrations = await this.getDecryptedIntegrationsUsecase.execute(
    GetDecryptedIntegrationsCommand.create({
        organizationId: command.organizationId, // I modified the code in previous PR to by default reach by org
        userId: command.userId,
        active: true,
    })
);

const isMultiProviderConfigurationEnabled = await this.getFeatureFlag.isMultiProviderConfigurationEnabled(
      FeatureFlagCommand.create({
        userId: command.userId,
        organizationId: command.organizationId,
        environmentId: command.environmentId,
      })
    );
    
 if (!isMultiProviderConfigurationEnabled) {
    organizationActiveIntegrations = organizationActiveIntegrations.filter(integration => integration.environmentId === command.environmentId);
 }
```
So I think it is viable to do unless I am missing any nuance, that could be possible. 😕

Reply:
We discussed it privately and we will follow @djabarovgeorge approach until we can untangle some changes that were implemented that block a simpler approach.